packages/client/src/context/Auth.context.tsx

```diff
@@ -106,28 +115,96 @@ export const AuthProvider: FC<AuthProviderProps> = ({ children }) => {
     handleUnauthenticated();
   };
 
+  // Handle switch tab
+  const handleTabChange = (_event: React.SyntheticEvent, tab: 'login' | 'signup' | 'reset') => {
+    setActiveTab(tab);
+  };
+
+  // Handle organization select
+  const handleOrganizationSelect = (event: SelectChangeEvent<string>) => {
+    const selectedOrganization = organizationList.find((org) => org.name === event.target.value);
+    setOrganization(selectedOrganization || null);
+  };
+
   return (
-    <AuthContext.Provider value={{ token, authenticated, decodedToken, logout }}>
-      {!authenticated && organization && (
-        <FirebaseLoginWrapper setToken={handleAuthenticated} organization={organization} />
-      )}
-      {authenticated && children}
-    </AuthContext.Provider>
+    <Container
```

Comment:
With the switch to have a lot of view logic in this component, its becoming difficult to reason through.

For example, you'll notice after you login that the tabs of the login logic are still visible over the rest of the webpage. I'd recommend either making an `Auth.component` to store the tab view or move the tab view into the Firebase wrapper.